cpp/tests/hashing/xxhash64_test.cpp

```diff
@@ -0,0 +1,170 @@
+                                                                    7292178400482025765ul,
+                                                                    2971168436322821236ul,
+                                                                    9380524276503839603ul,
+                                                                    9380524276503839603ul});
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(output->view(), expected);
+}
+
+TEST_F(HashXX64Test, TestDouble)
+{
+  auto col1 =
+    cudf::test::fixed_width_column_wrapper<double>{{-127.,
+                                                    -70000.125,
+                                                    0.0,
+                                                    200000.5,
+                                                    128.5,
+                                                    -0.0,
+                                                    std::numeric_limits<double>::infinity(),
+                                                    std::numeric_limits<double>::quiet_NaN(),
+                                                    std::numeric_limits<double>::max(),
+                                                    std::numeric_limits<double>::min(),
+                                                    std::numeric_limits<double>::lowest()}};
+
+  auto const output = cudf::hashing::xxhash64(cudf::table_view({col1}));
+
+  // these were generated using the CPU compiled version of the cuco xxhash64 source
+  // https://github.com/NVIDIA/cuCollections/blob/dev/include/cuco/detail/hash_functions/xxhash.cuh
+  auto expected = cudf::test::fixed_width_column_wrapper<uint64_t>({16892115221677838993ul,
+                                                                    1686446903308179321ul,
+                                                                    3803688792395291579ul,
+                                                                    18250447068822614389ul,
+                                                                    3511911086082166358ul,
+                                                                    4558309869707674848ul,
+                                                                    18031741628920313605ul,
+                                                                    16838308782748609196ul,
+                                                                    3127544388062992779ul,
+                                                                    1692401401506680154ul,
+                                                                    13770442912356326755ul});
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(output->view(), expected);
+}
+
+TEST_F(HashXX64Test, StringType)
+{
+  auto col1 = cudf::test::strings_column_wrapper(
+    {"The",
+     "quick",
+     "brown fox",
+     "jumps over the lazy dog.",
+     "I am Jack's complete lack of null value",
+     "A very long (greater than 128 bytes/char string) to test a a very long string",
```

Comment:
This is only 77 (or 75 if you remove the duplicate "a") chars long. Is that right?
```suggestion
     "A very long (greater than 128 bytes/char string) to test a very long string",
```